### PR TITLE
Auth2

### PR DESCRIPTION
**Pull request by carvilsi**
messaging for errors

### CODE REVIEW

File: test/WeaverUser.test.coffee [hunk 1]
```diff
@@ -114,3 +114,25 @@ describe 'Weaver User', ->
...
+    
+  it 'should fails trying to signUp with an existing userName', ->
+      user = new Weaver.User()
+      user.logIn('phoenix','phoenix')
+      .then(->
+        user.signUp('phoenix','andromeda','andromeda@univer.se','andromedas','SYSUNITE')
+      ).then(->
+        user.signUp('phoenix','andromeda','centaurus@univer.se','andromedas','SYSUNITE')
+      ).then().catch((error)->
+        assert.equal(error.code, WeaverError.DUPLICATE_VALUE)
```
Comment (gvdent) [inline test/WeaverUser.test.coffee]: This test only asserts that the specific error code is passed iff an error is given.

maybe add a ....then( ->
  assert(false)
) at the end so the test fails if no error is given.

File: test/WeaverUser.test.coffee [hunk 2]
```diff
@@ -114,3 +114,25 @@ describe 'Weaver User', ->
...
+      )
+
+    
+  it 'should fails trying to signUp with an existing userEmail', ->
+      user = new Weaver.User()
+      user.signUp('phoenix','andromedas','andromedas@univer.se','andromedas','SYSUNITE')
+      .then(->
+        user.signUp('phoenix','andro','andromedas@univer.se','andromedas','SYSUNITE')
+      ).then().catch((error)->
+        assert.equal(error.code, WeaverError.DUPLICATE_VALUE)
```
Comment (gvdent) [inline test/WeaverUser.test.coffee]: Same as previous comment

File: test/WeaverUser.test.coffee (hunk 1 above)
Comment (carvilsi) [inline test/WeaverUser.test.coffee]: 👍

File: test/WeaverUser.test.coffee (hunk 2 above)
Comment (carvilsi) [inline test/WeaverUser.test.coffee]: 👍

**Pull request merged**